addons/mail/static/src/models/autocomplete_input_view/autocomplete_input_view.js

```diff
@@ -98,59 +93,37 @@ registerModel({
-                this.discussOwnerAsChatMobileAddItemHeader ||
-                (
-                    this.discussSidebarCategoryOwnerAsAddingItemInput &&
-                    this.discussSidebarCategoryOwnerAsAddingItemInput.autocompleteMethod === 'chat'
-                )
+                this.discussViewOwnerAsSidebarAddChat ||
+                this.discussViewOwnerAsMobileAddChat ||
+                this.messagingMenuOwnerAsMobileNewMessageInput
             ) {
-                this.messaging.models['Partner'].imSearch({
-                    callback: (partners) => {
-                        const suggestions = partners.map(partner => {
-                            return {
-                                id: partner.id,
-                                value: partner.nameOrDisplayName,
-                                label: partner.nameOrDisplayName,
-                            };
-                        });
-                        res(_.sortBy(suggestions, 'label'));
-                    },
-                    keyword: req.term,
-                    limit: 10,
-                });
+                this._onSourceChat(req, res);
             }
             if (
-                this.discussOwnerAsChannelMobileAddItemHeader ||
-                (
-                    this.discussSidebarCategoryOwnerAsAddingItemInput &&
-                    this.discussSidebarCategoryOwnerAsAddingItemInput.autocompleteMethod === 'channel'
-                )
+                this.discussViewOwnerAsSidebarAddChannel ||
+                this.discussViewOwnerAsMobileAddChannel
             ) {
-                this.discussOwnerAsChannelMobileAddItemHeader.handleAddChannelAutocompleteSource(req, res);
+                this._onSourceChannel(req, res);
             }
         },
         /**
          * @private
-         * @returns {string|FieldCommand}
+         * @returns {string}
          */
         _computeCustomClass() {
-            if (this.discussSidebarCategoryOwnerAsAddingItemInput) {
-                if (this.discussSidebarCategoryOwnerAsAddingItemInput === this.messaging.discuss.categoryChannel) {
-                    return 'o_DiscussSidebarCategory_newChannelAutocompleteSuggestions';
-                }
-            }
-            if (this.messagingMenuOwnerAsMobileNewMessageInput) {
-                return 'o_MessagingMenu_mobileNewMessageInputAutocomplete';
-            }
+            return 'o_AutocompleteInputView_customClass';
         },
         /**
          * @private
          * @returns {boolean|FieldCommand}
          */
         _computeIsHtml() {
-            if (this.discussSidebarCategoryOwnerAsAddingItemInput) {
-                return this.discussSidebarCategoryOwnerAsAddingItemInput === this.messaging.discuss.categoryChannel;
+            if (
+                this.discussViewOwnerAsSidebarAddChannel ||
+                this.discussViewOwnerAsMobileAddChannel ||
+                this.messagingMenuOwnerAsMobileNewMessageInput
```

Comment:
This one is wrong and introducing security issue without escape in the user handlers.